ctapipe/ml/tools/apply.py

```diff
@@ -0,0 +1,255 @@
+                    self.log.warning(
+                        "No regressor model for telescope type %s, skipping tel %d",
+                        tel,
+                        tel_id,
+                    )
+                    self.apply_regressor = False
+                else:
+                    self.apply_regressor = True
+            if self._has_classifier:
+                if tel not in self.classifier.model.models:
+                    self.log.warning(
+                        "No classifier model for telescope type %s, skipping tel %d",
+                        tel,
+                        tel_id,
+                    )
+                    self.apply_classifier = False
+                else:
+                    self.apply_classifier = True
+
+            table = self.loader.read_telescope_events([tel_id])
+            if self.apply_regressor:
+                table.remove_columns(
+                    [c for c in table.colnames if c.startswith(regressor_prefix)]
+                )
+            if self.apply_classifier:
+                table.remove_columns(
+                    [c for c in table.colnames if c.startswith(classifier_prefix)]
+                )
+
+            if len(table) == 0:
+                self.log.warning("No events for telescope %d", tel_id)
+                continue
+
+            if self.apply_regressor:
+                predictions = self.regressor.predict(tel, table)
+                table = hstack(
+                    [table, predictions],
+                    join_type="exact",
+                    metadata_conflicts="ignore",
+                )
+                write_table(
+                    table[["obs_id", "event_id", "tel_id"] + predictions.colnames],
+                    self.loader.input_url,
+                    f"/dl2/event/telescope/energy/{regressor_prefix}/tel_{tel_id:03d}",
+                    mode="a",
+                    overwrite=self.overwrite,
+                )
+            if self.apply_classifier:
+                predictions = self.classifier.predict(tel, table)
+                table = hstack(
+                    [table, predictions],
+                    join_type="exact",
+                    metadata_conflicts="ignore",
+                )
+                write_table(
+                    table[["obs_id", "event_id", "tel_id"] + predictions.colnames],
+                    self.loader.input_url,
+                    f"/dl2/event/telescope/classification/{classifier_prefix}/tel_{tel_id:03d}",
+                    mode="a",
+                    overwrite=self.overwrite,
+                )
+            tables.append(table)
+
+        if len(tables) == 0:
+            raise ValueError("No predictions made for any telescope")
+
+        mono_predictions = vstack(tables)
+        if self.apply_regressor:
+            stereo_predictions = self.regressor_combine.predict(mono_predictions)
+            trafo = TelListToMaskTransform(self.loader.subarray)
```

Comment:
Probably we should add a `transformations=dict[str, Transform]` option to `write_table`, as it needs to include the metadata correctly (not for the tellist transform, but in general).